packages/wearable_rotary/tizen/src/wearable_rotary_plugin.cc

```diff
@@ -5,18 +5,80 @@
 #include "wearable_rotary_plugin.h"
 
 #include <efl-extension/wearable/circle/efl_extension_rotary.h>
-#include <flutter/encodable_value.h>
 #include <flutter/event_channel.h>
 #include <flutter/event_sink.h>
-#include <flutter/event_stream_handler_functions.h>
+#include <flutter/event_stream_handler.h>
 #include <flutter/plugin_registrar.h>
 #include <flutter/standard_method_codec.h>
 
 #include <memory>
+#include <optional>
+#include <string>
 
 #include "log.h"
 
-static constexpr char kChannelName[] = "flutter.wearable_rotary.channel";
+namespace {
+
+typedef flutter::EventChannel<flutter::EncodableValue> FlEventChannel;
+typedef flutter::EventSink<flutter::EncodableValue> FlEventSink;
+typedef flutter::StreamHandler<flutter::EncodableValue> FlStreamHandler;
+typedef flutter::StreamHandlerError<flutter::EncodableValue>
+    FlStreamHandlerError;
+
+class WearableRotaryStreamHandlerError : public FlStreamHandlerError {
+ public:
+  WearableRotaryStreamHandlerError(const std::string &error_code,
+                                   const std::string &error_message,
+                                   const flutter::EncodableValue *error_details)
+      : error_code_(error_code),
+        error_message_(error_message),
+        FlStreamHandlerError(error_code_, error_message_, error_details) {}
+
+  virtual ~WearableRotaryStreamHandlerError() {}
+
+ private:
+  std::string error_code_;
+  std::string error_message_;
+};
+
+class WearableRotaryStreamHandler : public FlStreamHandler {
+ public:
+  WearableRotaryStreamHandler() {}
+
+  virtual ~WearableRotaryStreamHandler() {}
+
+ protected:
+  std::unique_ptr<FlStreamHandlerError> OnListenInternal(
+      const flutter::EncodableValue *arguments,
+      std::unique_ptr<FlEventSink> &&events) override {
+    events_ = std::move(events);
+    Eina_Bool ret = eext_rotary_event_handler_add(RotaryEventCallBack, this);
+    if (ret == EINA_FALSE) {
+      return std::make_unique<WearableRotaryStreamHandlerError>(
+          std::to_string(TIZEN_ERROR_UNKNOWN),
```

Comment:
```suggestion
          "Operation failed",
```

Reply:
I applied it.

Reply:
No, you didn't :<

Reply:
oh, I misunderstood. this parameter is for error code not error description. why do you want to replace the error code with a message? (also I already search the error number for operation failed, but I couldn't find it in Tizen common error)

Reply:
I think you're misunderstanding the error code parameter. An error code is a string that indicates the type of the error, such as "Invalid argument" or "Permission denied" (see https://github.com/flutter-tizen/plugins/issues/354). It isn't necessarily a number defined in the Tizen API.

Reply:
ok I fixed it.